core-rust/core-api-server/src/core_api/conversions/receipt.rs

```diff
@@ -223,6 +232,39 @@ pub fn to_api_next_epoch(
     Ok(next_epoch)
 }
 
+#[tracing::instrument(skip_all)]
+pub fn to_api_event(
+    context: &MappingContext,
+    type_id: EventTypeIdentifier,
+    data: Vec<u8>,
+) -> Result<models::Event, MappingError> {
+    let EventTypeIdentifier(emitter, local_type_index) = type_id;
+    Ok(models::Event {
+        _type: Box::new(models::EventTypeIdentifier {
+            schema_local_index: match local_type_index {
+                LocalTypeIndex::SchemaLocalIndex(index) => index.try_into().unwrap(),
```

Comment:
_(note from author)_

I've discussed this with @0xOmarA and looks like we simply have to live with the fact that the `event`'s type is specified as the index of the type definition within the `emitter`'s schema. (please note it is never `WellKnown`, always `SchemaLocalIndex`, by design).

I've explored the ways to map the `index -> event name` here (following the https://github.com/radixdlt/radixdlt-scrypto/blob/0b0b30b04bd3f7c99ea608bf3425958f0fd32bda/scrypto-unit/src/test_runner.rs#L1139-L1222), and sadly, we don't contact the persistence in our API mapping layer. We could do it (preload schemas from the right blueprints?) in the handler, but it's rather cumbersome (and lots of code that would pollute this otherwise simple PR).

@KrzysztofLabus-RDX I imagine that the Gateway may be more interested in performing that mapping - please let me know if it becomes even more impractical on the Gateway's side.

Reply:
Right - I think the above approach is broadly correct - we can pass the event index upstream, and it'll be up to the Gateway API to resolve the correct schema at the given time when it responds to queries.

The Core API transaction stream operates on diffs, and I think it'd be wrong/incorrect to include stateful information there. (Although the package schema is - at least at the moment - immutable in the application layer, so this concern is a little less strong).

Reply:
Note that I added a LocalTypeIndex type into the schema that weshould possibly use for this 👍.

I don't think it necessarily matters at the Core API abstraction that the type can never be well known - I'm not even sure that that is an invariant to be honest(!).

EG there will be well known types which pass the struct/enum check in future! So I don't think we should panic here.

Reply:
awww the `LocalTypeIndex` api type did not exist when I started this PR - in this case, OFC, let me re-use it now 👍